tools/online_editor/src/properties_widget.ts

```diff
@@ -12,7 +12,20 @@ import { change_property, query_properties } from "./properties_client";
 import { Message } from "@lumino/messaging";
 import { Widget } from "@lumino/widgets";
 
-import { BaseLanguageClient } from "vscode-languageclient";
+import {
+    BaseLanguageClient,
+    HandleWorkDoneProgressSignature,
+    ProgressToken,
+    WorkDoneProgressBegin,
+    WorkDoneProgressEnd,
+    WorkDoneProgressReport,
+} from "vscode-languageclient";
+
+function extract_uri(input: string): string {
```

Comment:
Maybe this should be given a more specific name, such as extract_uri_from_progress_message?